server/src/main/java/org/apache/gravitino/server/web/rest/PermissionOperations.java

```diff
@@ -89,11 +101,19 @@ public Response grantRolesToGroup(
       return Utils.doAs(
           httpRequest,
           () ->
-              Utils.ok(
-                  new GroupResponse(
-                      DTOConverters.toDTO(
-                          accessControlManager.grantRolesToGroup(
-                              metalake, request.getRoleNames(), group)))));
+              TreeLockUtils.doWithTreeLock(
+                  NameIdentifier.of(AuthorizationUtils.ofGroupNamespace(metalake).levels()),
```

Comment:
Also here, group-level write lock should be enough.

Reply:
Done.